### PR TITLE
refactor: code review changes

### PR DESCRIPTION
**Pull request by bhumilsarvaiya**
## Change related to the following PR review:

- https://github.com/supertokens/supertokens-node/pull/274#discussion_r887807403
- https://github.com/supertokens/supertokens-node/pull/274#discussion_r887888724
- updated email templates

## Related PR

-   #274

### CODE REVIEW

File: lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts [hunk 1]
```diff
@@ -47,7 +46,7 @@ export default class BackwardCompatibilityService
         let emailVerificationFeatureNormalisedConfig =
             inputCreateAndSendCustomEmail !== undefined
                 ? {
-                      createAndSendCustomEmail: async (user: EmailVerificationUser, link: string, userContext: any) => {
+                      createAndSendCustomEmail: async (user: EmailPasswordUser, link: string, userContext: any) => {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts:49]: You need to build project as well

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts (hunk 1 above)
Comment (bhumilsarvaiya) [reply lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts:49]: I did, no changes

Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts:49]: - Add a test to sign up with third party and then send them an email verification email (this should fail)
- We have to create an email verification backwards compatibility service as well (and then the above test should pass).

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: test/thirdpartyemailpassword/emailverify.test.js [hunk 2]
```diff
@@ -239,6 +268,68 @@ describe(`emailverify: ${printPath("[test/thirdpartyemailpassword/emailverify.te
         assert(emailToken !== null);
     });
 
+    it("test that providing your own email callback and make sure it is called (thirdparty user)", async function () {
```
Comment (rishabhpoddar) [inline test/thirdpartyemailpassword/emailverify.test.js:271]: We need to add more tests like:
- Emailpassword recipe:
   - creating email password user and doing reset password and email verification email
   - Trying to send reset password to non existent user should not call your callback
- Third party recipe:
   - creating third party user and sending email verification to them.
- Third party email password recipe:
   - creating email password user and doing reset password and email verification email
   - creating third party user and doing email verification email
   - creating third party user and doing password reset -> should not call the callback
   - Trying to send reset password to non existent user should not call your callback.
- Passwordless recipe:
   - Sending email and SMS for login should call callback
-  Third party passwordless recipe:
   - Sending email and SMS for passwordless login should call callback
   - Email verification email for third party user -> should call callback
   - Email verification for passwordless user -> should not call callback

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**